openfeature/provider.go

```diff
@@ -186,7 +186,7 @@ type InterfaceResolutionDetail struct {
 	ProviderResolutionDetail
 }
 
-// Metadata provides provider name
+// Metadata provides provider domain
 type Metadata struct {
-	Name string
+	Domain string
 }
```

Comment:
This should be revered :) Provider name is still valid and it's Provider's name (ex:- flagd provider https://github.com/open-feature/go-sdk-contrib/blob/providers/flagd/v0.1.22/providers/flagd/pkg/provider.go#L144-L148 )